perun-oidc-server/src/main/java/cz/muni/ics/oauth2/web/OAuthConfirmationController.java

```diff
@@ -114,36 +155,82 @@ public String confirmAccess(Map<String, Object> model, Principal p) {
 			return HttpCodeView.VIEWNAME;
 		}
 
-		if (prompts.contains("none")) {
+		if (prompts.contains(NONE)) {
 			// if we've got a redirect URI then we'll send it
+			return sendRedirect(authRequest, model, client);
+		}
 
-			String url = redirectResolver.resolveRedirect(authRequest.getRedirectUri(), client);
+		model.put(AUTH_REQUEST, authRequest);
+		model.put(CLIENT, client);
 
-			try {
-				URIBuilder uriBuilder = new URIBuilder(url);
+		String redirect_uri = authRequest.getRedirectUri();
+		model.put(REDIRECT_URI, redirect_uri);
 
-				uriBuilder.addParameter("error", "interaction_required");
-				if (!Strings.isNullOrEmpty(authRequest.getState())) {
-					uriBuilder.addParameter("state", authRequest.getState()); // copy the state parameter if one was given
-				}
+		Set<SystemScope> sortedScopes = getSortedScopes(authRequest);
+		model.put(SCOPES, sortedScopes);
 
-				return "redirect:" + uriBuilder;
+		// get the userinfo claims for each scope
+		Map<String, Map<String, String>> claimsForScopes = getClaimsForScopes(p, sortedScopes);
+		model.put(CLAIMS, claimsForScopes);
 
-			} catch (URISyntaxException e) {
-				log.error("Can't build redirect URI for prompt=none, sending error instead", e);
-				model.put("code", HttpStatus.FORBIDDEN);
-				return HttpCodeView.VIEWNAME;
-			}
+		// contacts
+		if (client.getContacts() != null) {
+			String contacts = Joiner.on(", ").join(client.getContacts());
+			model.put(CONTACTS, contacts);
+		}
+		model.put(GRAS, true);
+
+		if (!perunOidcConfig.getTheme().equalsIgnoreCase(DEFAULT)) {
+			model.remove(SCOPES);
+			model.remove(CLAIMS);
+			PerunUserInfo perunUser = (PerunUserInfo) userInfoService.getByUsernameAndClientId(
+					p.getName(), client.getClientId());
+			ControllerUtils.setScopesAndClaims(scopeService, perunScopeClaimTranslationService, model, authRequest.getScope(),
```

Comment:
Try to check if the static method could be refactored to accept the generic `ScopeClaimTranslationService` parameter instead of tying it specifically to `PerunScope...` implementation.